data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -181,4 +188,169 @@ private TableMigrationUtil() {
+                  .map(PartitionField::name)
+                  .map(name -> String.format("%s=%s", name, partitionPath.get(name)))
+                  .collect(Collectors.joining("/"));
+
+          return DataFiles.builder(spec)
+                  .withPath(stat.getPath().toString())
+                  .withFormat("parquet")
+                  .withFileSizeInBytes(stat.getLen())
+                  .withMetrics(metrics)
+                  .withPartitionPath(partitionKey)
+                  .build();
+        });
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to list files in partition: " + partitionUri, e);
+    }
+
+    try {
+      return runAllInParallel(parallelism, footers);
+    } catch (ExecutionException e) {
+      throw new RuntimeException("Could not read footer: " + e.getMessage(), e.getCause());
+    }
+
+  }
+
+  private static List<DataFile> listOrcPartition(Map<String, String> partitionPath, String partitionUri,
+                                                     PartitionSpec spec, Configuration conf,
+                                                     MetricsConfig metricsSpec, NameMapping mapping, int parallelism) {
+    // not use parallelism
+    if (parallelism == 0 || parallelism == 1) {
+      return listOrcPartition(partitionPath, partitionUri, spec, conf, metricsSpec, mapping);
+    }
+
+    List<Callable<DataFile>> footers = new ArrayList<>();
+    try {
+      Path partition = new Path(partitionUri);
+      FileSystem fs = partition.getFileSystem(conf);
+      FileStatus[] fileStatus = fs.listStatus(partition, HIDDEN_PATH_FILTER);
+      for (final FileStatus stat : fileStatus) {
+        if (!stat.isFile()) {
+          continue;
+        }
+        footers.add(() -> {
+          Metrics metrics = OrcMetrics.fromInputFile(HadoopInputFile.fromPath(stat.getPath(), conf),
+                  metricsSpec, mapping);
+          String partitionKey = spec.fields().stream()
+                  .map(PartitionField::name)
+                  .map(name -> String.format("%s=%s", name, partitionPath.get(name)))
+                  .collect(Collectors.joining("/"));
+
+          return DataFiles.builder(spec)
+                  .withPath(stat.getPath().toString())
+                  .withFormat("orc")
+                  .withFileSizeInBytes(stat.getLen())
+                  .withMetrics(metrics)
+                  .withPartitionPath(partitionKey)
+                  .build();
+        });
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Unable to list files in partition: " + partitionUri, e);
+    }
+
+    try {
+      return runAllInParallel(parallelism, footers);
+    } catch (ExecutionException e) {
+      throw new RuntimeException("Could not read footer: " + e.getMessage(), e.getCause());
+    }
+
+  }
+
+  private static <T> List<T> runAllInParallel(int parallelism, List<Callable<T>> toRun) throws ExecutionException {
```

Comment:
Rather than using a new method, can you use the `Tasks` utility?

Reply:
Thansk for your  review,  because I' m a newer to java, Can u give me a link about the `Tasks` utility , so I can learn how to replace it with `Tasks` .

Reply:
Yea I think that's what is meant here